ddtrace/tracer/option.go

```diff
@@ -547,7 +547,9 @@ func newConfig(opts ...StartOption) *config {
 		if agentport := c.agent.StatsdPort; agentport > 0 && !c.agent.ignore {
 			// the agent reported a non-standard port
 			host, _, err := net.SplitHostPort(addr)
-			if err == nil {
+			// the code in the next if-block only makes sense if we are dealing with
+			// a user-defined TCP-based protocol URI.
+			if err == nil && host != "unix" {
```

Comment:
Is `host == unix` the only case in which we want to ignore the agent's reported statsd port?

I would think either we universally respect the value provided by the user in `WithDogStatsdAddr` or we universally use the agent's reported address.